sstables_loader.cc

```diff
@@ -258,14 +265,26 @@ future<> tablet_sstable_streamer::stream(std::function<void(unsigned)> on_stream
             co_await coroutine::maybe_yield();
         }
 
+        if (on_streamed) {
+            double progress_unit = 1. / _tablet_map.tablet_count();
+            size_t num_sstables = sstables_fully_contained.size() + sstables_partially_contained.size();
+            if (num_sstables > 0) {
+                on_streamed = [unit = progress_unit / num_sstables, on_streamed = std::move(on_streamed)] (double processed, double) {
+                    std::invoke(on_streamed, processed * unit, 0.);
```

Comment:
This updates progress with the value of
```
processed * unit = processed * progress_unit / num_sstables = processed * (1 / _tablet_map.tablet_count()) / all_sstables.size() = processed / (_tablet_map.tablet_count() * all_sstables.size())
```

Next, `tablet_sstable_streamer::stream()` sets "total" to be `_tablet_map.tablet_count()`

Finally, the implementation of on_streamed() does this:

```
   progress.processed += processed;
   progress.total = std::max(progress.total, total);
```

In the end of the day `progress.total = _tablet_map.tablet_count()` and `progress.processed = all_sstables.size() * _tablet_map.count()`

Please, correct my understanding

Reply:
> This updates progress with the value of
> 
> ```
> processed * unit = processed * progress_unit / num_sstables = processed * (1 / _tablet_map.tablet_count()) / all_sstables.size() = processed / (_tablet_map.tablet_count() * all_sstables.size())
> ```
> 
> Next, `tablet_sstable_streamer::stream()` sets "total" to be `_tablet_map.tablet_count()`
> 
> Finally, the implementation of on_streamed() does this:
> 
> ```
>    progress.processed += processed;
>    progress.total = std::max(progress.total, total);
> ```
> 
> In the end of the day `progress.total = _tablet_map.tablet_count()` and `progress.processed = all_sstables.size() * _tablet_map.count()`
> 

i think the bug in previous revision is that, 

1. i should not advance the progress in the inner loop by "batch_sst_nr / sstables" -- i should have just call `on_stream(batch_sst_nr)", as i've already divided it by "sstables_partially_contained.size() + sstables_fully_contained.size()"
2. i should not divide the unit by `_tablet_map.tablet_count()`. because the unit should be "1". and the sum of all the units are ` _tablet_map.tablet_count()`

anyway, the math is confusing. so i reworked it by introducing another progress class for tracking the per-tablet progress.